javascript/react/src/components/Codebook/Codebook.styles.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { IDropdownStyles, ITextFieldStyles, Theme } from '@fluentui/react'
+import { useTheme } from '@fluentui/react'
+import type { CSSProperties } from 'react'
+import { useMemo } from 'react'
+
+import type { FieldHeights } from './Codebook.hooks.js'
+import type { CodebookDefaultStyles } from './Codebook.types.js'
+
+export function getRootStyle(
+	theme: Theme,
+	disabled?: boolean,
+	style?: CSSProperties,
+): CSSProperties {
+	return {
+		backgroundColor: disabled
+			? theme.palette.neutralLighter
+			: theme.palette.white,
+		...style,
+	}
+}
+
+export function useDefaultStyles(
+	styles?: CodebookDefaultStyles,
+	heights?: FieldHeights,
+): CodebookDefaultStyles {
+	const theme = useTheme()
+	return useMemo(
+		() => ({
+			mapping: {
+				root: {
+					height: heights?.get('mappingWrapper'),
+					...styles?.mapping?.root,
```

Comment:
once you start getting multiple nested merged, the lodash `merge` utility becomes a lot cleaner to use